pyquil/quil.py

```diff
@@ -644,30 +671,28 @@ def _synthesize(self):
         self._synthesized_instructions = implicitly_declare_ro(self._synthesized_instructions)
         return self
 
-    def __add__(self, other):
+    def __add__(self, other: "Program") -> "Program":
         """
         Concatenate two programs together, returning a new one.
 
-        :param Program other: Another program or instruction to concatenate to this one.
+        :param other: Another program or instruction to concatenate to this one.
         :return: A newly concatenated program.
-        :rtype: Program
         """
         p = Program()
         p.inst(self)
         p.inst(other)
         return p
 
-    def __iadd__(self, other):
+    def __iadd__(self, other: object) -> "Program":
         """
         Concatenate two programs together using +=, returning a new one.
 
-        :param Program other: Another program or instruction to concatenate to this one.
+        :param other: Another program or instruction to concatenate to this one.
         :return: A newly concatenated program.
-        :rtype: Program
         """
         return self.inst(other)
 
-    def __getitem__(self, index):
+    def __getitem__(self, index: Any) -> "Program":
```

Comment:
```suggestion
    def __getitem__(self, index: Any) -> Union[AbstractInstruction, "Program"]:
```

Also, is `index` really `Any` here?  Perhaps `Union[slice, int]`?

Reply:
Used `Union[slice, int]` in 5th commit.